esmvalcore/_data_finder.py

```diff
@@ -156,16 +160,26 @@ def _resolve_latestversion(dirname_template):
     This implementation avoid globbing on centralized clusters with very
     large data root dirs (i.e. ESGF nodes like Jasmin/DKRZ).
     """
-    if '{latestversion}' not in dirname_template:
+    if "{latestversion}" not in dirname_template:
         return dirname_template
 
     # Find latest version
-    part1, part2 = dirname_template.split('{latestversion}')
+    part1, part2 = dirname_template.split("{latestversion}")
+    # resolve any wildcards entered for fx variables
+    if "/fx/" in part1:
+        dirs = glob.glob(part1)
```

Comment:
ah very nice, I just wrote my comment to the review post without seeing the new changes :+1:

Reply:
yeah I just fixed those tests, @schlunma would you mind having a look too, these are tests that have been freshly added from `master` after a previous merge of your data finder test improving branch :+1: :beer:

Reply:
cheers for willing to do dox @thomascrocker :beer:

Reply:
I'm not really convinced of this solution:

1. Sometimes fx variables for CMIP6 are stored in other tables, e.g. `Omon` (if the files are time dependent) or `Ofx`. Both cases are not considered here.
2. Why would we ignore subsequent directories if more than one is found?

Reply:
I'm still not convinced by this solution. Checking paths directly just doesn't feel right (e.g. I guess if we ever support Windows, this wouldn't work for paths that use `\` instead of `/`). Moreover, I find it weird to put that code into the function `_resolve_latestversion`.

As an alternative, how about you write a dedicated function `_resolve_fx_wildcards` (or a similar name) and put it right before the [call to `_resolve_latestversion`](https://github.com/ESMValGroup/ESMValCore/blob/87b5f489c6aa10d168e8924af521379fb10fb260/esmvalcore/_data_finder.py#L235)? You can pass the argument `variable` to this function, which can then be checked for `variable['frequency'] == 'fx'`. I think this is way safer than the current check of the path.

Reply:
Thanks for the suggestion, I have refactored my approach using a new function as you suggested. I've also added a couple of tests for CMIP6 and CORDEX using the wildcard functionality. The original FX tests remain in place and the code passes them also. Let me know what you think.